source/JsMaterialX/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+set(OUTPUT_CORE ${CMAKE_CURRENT_SOURCE_DIR}/JsMaterialXCore/)
+set(OUTPUT_FORMAT ${CMAKE_CURRENT_SOURCE_DIR}/JsMaterialXFormat/)
+set(OUTPUT_LIBS ${PROJECT_BINARY_DIR}/source/)
+
+set(SOURCE_FOLDER ${CMAKE_SOURCE_DIR}/source)
+
+set(CORE_DEPS ${OUTPUT_CORE}JsTypes.cpp
+    ${OUTPUT_CORE}JsExceptions.cpp
+    ${OUTPUT_CORE}JsUtil.cpp
+    ${OUTPUT_CORE}JsElement.cpp
+    ${OUTPUT_CORE}JsInterface.cpp
+    ${OUTPUT_CORE}JsNode.cpp
+    ${OUTPUT_CORE}JsDefinition.cpp
+    ${OUTPUT_CORE}JsDocument.cpp
+    ${OUTPUT_CORE}JsLook.cpp
+    ${OUTPUT_CORE}JsProperty.cpp
+    ${OUTPUT_CORE}JsGeom.cpp
+    ${OUTPUT_CORE}JsVariant.cpp
+    ${OUTPUT_CORE}JsValue.cpp
+    ${OUTPUT_CORE}JsMaterial.cpp
+    ${OUTPUT_CORE}JsRegisteredVectors.cpp
+    ${OUTPUT_CORE}JsTraversal.cpp
+    ${OUTPUT_CORE}JsObserver.cpp)
+
+set(FORMAT_DEPS ${OUTPUT_FORMAT}JsXmlIo.cpp)
+
+set(LIB_DEPS ${OUTPUT_LIBS}MaterialXCore/libMaterialXCore.a ${OUTPUT_LIBS}MaterialXFormat/libMaterialXFormat.a)
+
+# Include the source directory
+set(JS_FLAGS "-I${SOURCE_FOLDER} ")
+
+# Add all of the required flags for emcc
+string(APPEND JS_FLAGS "-std=c++17 -s WASM=1 -s DISABLE_EXCEPTION_CATCHING=0 -s MODULARIZE=1 -s EXPORT_ES6=1 -s USE_ES6_IMPORT_META=0 -s MEMORY_GROWTH_LINEAR_STEP=32MB -s ALLOW_MEMORY_GROWTH=1 ")
```

Comment:
Minor: Leaving comment from before that this should be key'ed off of the cmake build option so we can build other version as required.